store/postgres/src/query_store.rs

```diff
@@ -0,0 +1,48 @@
+use std::collections::BTreeMap;
+
+use crate::store::ReplicaId;
+use graph::components::store::QueryStore as QueryStoreTrait;
+use graph::prelude::{Store as _, *};
+
+pub(crate) struct QueryStore {
+    replica_id: ReplicaId,
+    store: Arc<crate::Store>,
+    for_subscription: bool,
+}
+
+impl QueryStore {
+    pub(crate) fn new(store: Arc<crate::Store>, for_subscription: bool) -> Self {
+        // Subscriptions always go to the main replica.
+        let replica_id = match for_subscription {
+            false => store.pick_replica(),
```

Comment:
I find this back-and-forth between `Store` and `QueryStore` a little strange; it would be more straightforward to make `QueryStore::new` part of `Store.pick_replica` so that this whole logic is in one place. All of this logic could probably go into `Store::query_store`

Reply:
You're saying we can remove `QueryStore::new` by inlining it in `Store::query_store`? Makes sense. I suppose I can then also do the same with `pick_replica`.